ansible/configs/ocp4-cluster/lifecycle_hook_post_start.yml

```diff
@@ -79,3 +79,43 @@
       when: r_new_csrs.resources | length > 0
       command: "oc adm certificate approve {{ item.metadata.name }}"
       loop: "{{ r_new_csrs.resources }}"
+
+    # Get OpenShift Version to determine if we need additional recovery actions
+    - name: Get OpenShift ClusterVersion
+      k8s_facts:
+        api_version: config.openshift.io/v1
+        kind: ClusterVersion
+        name: version
+      register: r_cluster_version
+    - name: Set ocp4_cluster_version fact
+      set_fact:
+        ocp4_cluster_version: "{{ r_cluster_version.resources[0].status.history[0].version }}"
+
+    # Only do the following recovery actions on 4.4.7 and higher 4.4 OpenShift versions
+    # Before 4.4.7 the cluster would still be broken. Starting in 4.5 this won't be necessary
+    # Also only do when previously there were certificates to be approved
+    - name: Workaround for OCP 4.4 bug (API Servers need restarting)
+      when:
+      - ocp4_cluster_version is version_compare('4.4.7', '>=')
+      - ocp4_cluster_version is version_compare('4.5.0', '<')
+      - r_csrs.resources | length > 0 or r_new_csrs.resources | length > 0
+      block:
+      - name: Wait 5 minutes for things to settle
```

Comment:
Is there something else we can look at here? I don't like picking a number like this and hoping it is settled...what if it takes longer for some reason (we've run into this often enough before)? Is there something that can be identified as "settled"?

Reply:
Not really. Unless you want to start parsing COs status which is super messy.

Reply:
Something like?
```
  - name: Wait for cluster operators to settle
    k8s_info:
      api_version: config.openshift.io/v1
      kind: ClusterOperator
    register: r_get_co
    failed_when: >-
      r_get_co | json_query(_condition_unavailable_query) | length > 0 or
      r_get_co | json_query(_condition_progressing_query) | length > 0 or
      r_get_co | json_query(_condition_degraded_query) | length > 0
    vars:
      _condition_unavailable_query: >-
        resources[].status.conditions[?type=='Available' && status=='False'][]
      _condition_progressing_query: >-
        resources[].status.conditions[?type=='Progressing' && status=='True'][]
      _condition_degraded_query: >-
        resources[].status.conditions[?type=='Degraded' && status=='True'][]
    until: r_get_co is success
    retries: 100
    delay: 30
```

Reply:
I think we may not need this at all anymore. Will know Monday.